src/cmap/connect.ts

```diff
@@ -214,7 +214,7 @@ export async function prepareHandshakeDocument(
   const { serverApi } = authContext.connection;
 
   const handshakeDoc: HandshakeDocument = {
-    [serverApi?.version ? 'hello' : LEGACY_HELLO_COMMAND]: 1,
+    [serverApi?.version || options.loadBalanced === true ? 'hello' : LEGACY_HELLO_COMMAND]: 1,
```

Comment:
```ts
[serverApi?.version || options.loadBalanced ? 'hello' : LEGACY_HELLO_COMMAND]: 1,
```

Is there something specific we are looking for by explicitly checking for `true`? I've seen reasons to check for precisely `false`, to differentiate falsey from truly `false` but not the other way around 🤔

Reply:
A bit lower in the code we explicitly check for `true` to then set the `handshakeDoc.loadBalanced` value:
https://github.com/mongodb/node-mongodb-native/blob/NODE-3881-hello-plus-op_msg/src/cmap/connect.ts#L223

And the [spec](https://github.com/mongodb/specifications/blob/master/source/mongodb-handshake/handshake.rst) says `If a server API version is requested or loadBalanced: True`. I feel like this is a matching behaviour in this case.